src/Duende.Bff/PostConfigureOidcOptionsForSilentLogin.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) Duende Software. All rights reserved.
+// See LICENSE in the project root for license information.
+
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Threading.Tasks;
+
+namespace Duende.Bff
+{
+    /// <summary>
+    /// OIDC configuration to add silent login support
+    /// </summary>
+    public class PostConfigureOidcOptionsForSilentLogin : IPostConfigureOptions<OpenIdConnectOptions>
+    {
+        private readonly string? _scheme;
+        private readonly ILogger<PostConfigureOidcOptionsForSilentLogin> _logger;
+
+        /// <summary>
+        /// ctor
+        /// </summary>
+        /// <param name="authenticationSchemeProvider"></param>
+        /// <param name="logger"></param>
+        public PostConfigureOidcOptionsForSilentLogin(IAuthenticationSchemeProvider authenticationSchemeProvider, ILogger<PostConfigureOidcOptionsForSilentLogin> logger)
+        {
+            _scheme = authenticationSchemeProvider.GetDefaultChallengeSchemeAsync().Result?.Name;
+            _logger = logger;
+        }
+
+        /// <inheritdoc />
+        public void PostConfigure(string name, OpenIdConnectOptions options)
+        {
+            if (name == _scheme)
+            {
+                options.Events.OnRedirectToIdentityProvider = CreateRedirectCallback(options.Events.OnRedirectToIdentityProvider);
```

Comment:
This will not work if the `Options.EventType` property is used to get the Events instance from DI, see https://github.com/dotnet/aspnetcore/blob/c911002ab43b7b989ed67090f2a48d9073d5118d/src/Security/Authentication/Core/src/AuthenticationHandler.cs#L149